Makefile

```diff
@@ -74,121 +82,62 @@ emulate:
 	firebase emulators:start --only hosting --project ${FIREBASE_ALIAS}
 
 
-# =================== Testing locally from host ==================
-# NOTE that these are for convenience of testing from outside of a container. 
-# You could just as easily enter a fresh container shell and run these as any 
-# code changes on host would be reflected inside the container. e.g. 
-# ```
-# docker-compose run --rm site bash
-# $ make switch-channel CHANNEL=dev
-# $ tool/test.sh --target ${CHANNEL} --check-links --null-safety
-# ```
-CHANNEL ?= stable
-
-# Convenience command to run inside of a container to 
-# switch the channel of the flutter submodule. 
-# NOTE that changes will be reflected in your local flutter repo 
-# Usage: `make switch-channel CHANNEL=<channel>`
-switch-channel:
-	cd flutter && \
-		git remote set-branches origin ${CHANNEL} && \
-  		git fetch --depth 1 origin ${CHANNEL} && \
-  		git checkout ${CHANNEL} --  && \
-  		git pull && \
-		cd .. && \
-  		flutter doctor
-
-# Run tests against a channel
-# NOTE this will mirror test artifacts to local via the volume
-# Usage: `make test-channel CHANNEL=<channel>`
-test-channel:
-	docker-compose run -d --rm site
-	docker-compose exec site tool/test.sh --target ${CHANNEL} --check-links --null-safety
-	docker-compose stop site
-
-# Test refreshing code excerpts
-# NOTE this will mirror test artifacts to local via the volume
-# Usage: `make refresh-code-excerpts`
-refresh-code-excerpts:
-	docker-compose run -d --rm site
-	docker-compose exec site tool/refresh-code-excerpts.sh --keep-dart-tool
-	docker-compose stop site
-
-# Test checking code snippets
-# NOTE this will mirror test artifacts to local via the volume
-# Usage: `make check-code`
-check-code:
-	docker-compose run -d --rm site
-	docker-compose exec site tool/check-code.sh --refresh --null-safety
-	docker-compose stop site
-
-# Test checking links
-# NOTE this will mirror test artifacts to local via the volume
-# Usage: `make check-links`
-check-links:
-	docker-compose run -d --rm site
-	docker-compose exec site tool/check-links.sh
-	docker-compose stop site
-
-
-
 # =================== Build / Test / Deploy Commands ==================
 
-FLUTTER_BRANCH ?= stable
-TEST_TARGET_CHANNEL ?= stable
-BUILD_CONFIGS ?= _config.yml
-BUILD_COMMIT := $(shell git rev-parse --short HEAD)
-BUILD_TAG = fltbuild
-BUILD_NAME = tmpbuild
-
 # Run all tests that would be run inside the container for the 
 # given target channel on the Github action `test` job.
-# WARNING this can take long while to run!
-# # Usage: `make test TEST_TARGET_CHANNEL=<channel>`
+# WARNING this can take a while to run!
+# Usage: `make test FLUTTER_TEST_BRANCH=<channel>`
 test:
-	docker build --rm --target test \
-		--build-arg TEST_TARGET_CHANNEL=${TEST_TARGET_CHANNEL} \
-		--build-arg RUBY_VERSION=${RUBY_VERSION} \
-		--build-arg NODE_VERSION=${NODE_VERSION} -t ${BUILD_TAG}:${BUILD_COMMIT} .
+	DOCKER_BUILDKIT=1 time docker build --rm --target tests -t flt-test .
```

Comment:
time is so helpful! I'm thinking we can leave it even after improving the optimizations are implemented. It'll give us a good baseline in case the build time starts increasing again.